src/protocol/vm/errors.rs

```diff
@@ -6,46 +6,7 @@ use ethers::prelude::ProviderError;
 use serde_json::Error as SerdeError;
 use thiserror::Error;
 
-use crate::evm::simulation::SimulationError;
-
-/// Represents the outer-level, user-facing errors of the tycho-simulation package.
-///
-/// `TychoSimulationError` encompasses all possible errors that can occur in the package,
-/// wrapping lower-level errors in a user-friendly way for easier handling and display.
-///
-/// Variants:
-/// - `AbiError`: Represents an error when loading the ABI file, encapsulating a `FileError`.
-/// - `EncodingError`: Denotes an error in encoding data.
-/// - `SimulationFailure`: Wraps errors that occur during simulation, containing a
-///   `SimulationError`.
-/// - `DecodingError`: Indicates an error in decoding data.
-/// - `RPCError`: Indicates an error related to RPC interaction.
-/// - `UnsupportedCapability`: Denotes an error when a pool state does not support a necessary
-///   capability.
-/// - `UninitializedAdapter`: Indicates an error when trying to use the Adapter before initializing
-///   it.
-/// - `CapabilityRetrievalFailure`: Indicates an error when trying to retrieve capabilities.
-/// - `EngineNotSet`: Indicates an error when trying to use the engine before setting it.
-//   the adapter.
-#[derive(Error, Debug)]
-pub enum TychoSimulationError {
-    #[error("ABI loading error: {0}")]
-    AbiError(FileError),
-    #[error("Encoding error: {0}")]
-    EncodingError(String),
-    #[error("Simulation failure error: {0}")]
-    SimulationFailure(SimulationError),
-    #[error("Decoding error: {0}")]
-    DecodingError(String),
-    #[error("RPC related error {0}")]
-    RpcError(RpcError),
-    #[error("Unsupported Capability: {0}")]
-    UnsupportedCapability(String),
-    #[error("Adapter not initialized: {0}")]
-    UninitializedAdapter(String),
-    #[error("Engine not set")]
-    EngineNotSet(),
```

Comment:
Pretty much all of these except `SimulationFailure` seem to be Fatal so you can map them to `Unkown()`